### PR TITLE
use memio in d_deh.c

### PR DESCRIPTION
**Pull request by rfomin**
Fixes ASan report.

### CODE REVIEW

File: src/d_deh.c [hunk 1]
```diff
@@ -1574,22 +1596,25 @@ void ProcessDehFile(const char *filename, char *outfilename, int lumpnum)
...
         }
+
+      infile.stream = mem_fopen_read(buf, W_LumpLength(lumpnum));
+      filename = W_WadNameForLump(lumpnum);
+
+      infile.file = NULL;
     }
 
-  printf("Loading DEH %sfile %s\n", infile.lump ? "lump from " : "", filename);
+  printf("Loading DEH %sfile %s\n", infile.file ? "lump from " : "", filename);
```
Comment (fabiangreffrath) [inline src/d_deh.c:1617]: This seems wrong, "lump from file" should report a PWAD.

File: src/d_deh.c [hunk 2]
```diff
@@ -43,61 +44,82 @@ static boolean bfgcells_modified = false;
...
+  MEMFILE *stream;
+  FILE *file;
 } DEHFILE;
 
 // killough 10/98: emulate IO whether input really comes from a file or not
 
 // haleyjd: got rid of macros for MSCV
 
-char *dehfgets(char *buf, size_t n, DEHFILE *fp)
+char *dehfgets(char *str, size_t count, DEHFILE *fp)
```
Comment (fabiangreffrath) [inline src/d_deh.c:55]: I guess we can turn all these functions into `static`?

File: src/d_deh.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/d_deh.c:1617]: Wait, I think I misread the diff.

**Review by fabiangreffrath — APPROVED** [review]

File: src/d_deh.c (hunk 1 above)
Comment (rfomin) [reply src/d_deh.c]: I think you are right, I fixed it and renamed `stream` to `lump`: https://github.com/fabiangreffrath/woof/pull/1031/commits/caf08bc71ae6b92b4b4fa0cdf8467947e5c807c3

File: src/d_deh.c (hunk 2 above)
Comment (rfomin) [reply src/d_deh.c]: Done. BTW we can add `static` to a lot of functions, should I do it?

File: src/d_deh.c [hunk 3]
```diff
@@ -34,6 +34,7 @@
 #include "d_think.h"
 #include "d_main.h" // D_DehChangePredefinedTranslucency()
 #include "w_wad.h"
+#include "memio.h"
```
[the remark this answers is not in the document]
Comment (rfomin) [reply src/d_deh.c:37]: Yes, I think so.

File: src/d_deh.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/d_deh.c]: Thanks, it was a bit confusing with the rename. 👍

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**